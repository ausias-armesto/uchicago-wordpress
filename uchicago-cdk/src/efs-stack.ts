import { Stack, Construct, Tags } from '@aws-cdk/core'
import { IVpc, Peer, Port, SecurityGroup, SubnetType, Vpc } from '@aws-cdk/aws-ec2'
import { FileSystem, LifecyclePolicy, PerformanceMode, OutOfInfrequentAccessPolicy, AccessPoint } from '@aws-cdk/aws-efs';
import { WordpressStackProps } from './wordpress-stack-props';

/**
 * Custom Properties for the EFS Stack
 */
 export interface EFSStackProps extends WordpressStackProps {
}

/**
 * Construct EFS File System resource on AWS
 */
export class EFSStack extends Stack {

  private props: EFSStackProps;
  private vpc: IVpc;
  public fileSystem: FileSystem;
  public securityGroup: SecurityGroup;
  private accessPoint: AccessPoint;

  constructor(scope: Construct, id: string, props?: EFSStackProps) {
    super(scope, id, props)
    this.props = props!;

    this.vpc = Vpc.fromLookup(this, 'vpc', { vpcName: 'vpc-' + this.props.environmentName })

    this.createSecurityGroup()
    this.createEFS()
  }

  private createSecurityGroup(): void {
    const securityGroupName = `secg-efs-${this.props.environmentName}`
    this.securityGroup = new SecurityGroup(this,securityGroupName,{ 
      vpc: this.vpc,
      securityGroupName,
      description: 'Security Group for EFS'
     })
     this.securityGroup.addIngressRule(Peer.ipv4('10.0.0.0/16'), Port.tcp(2049), 'Allow NFS access within the VPC')
  }

  /**
   * Creates a NFS Share
   * @returns 
   */
  private createEFS(): void {
    const fileSystemName = `efs-${this.props.environmentName}`
    this.fileSystem = new FileSystem(this, fileSystemName, {
      vpc: this.vpc,
      fileSystemName,
      encrypted: true,
      securityGroup: this.securityGroup,
      vpcSubnets: {
        subnetType: SubnetType.PRIVATE_WITH_NAT
      }
    });
    //this.accessPoint = this.fileSystem.addAccessPoint('RootAccessPoint');
    //Tags.of(this.accessPoint).add('Name','RootAccessPoint');
  }

  

}
