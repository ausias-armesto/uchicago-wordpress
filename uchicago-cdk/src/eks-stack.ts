import { Stack, App, Duration } from '@aws-cdk/core';
import { Vpc, SubnetType, InstanceType, SecurityGroup, Peer, Port } from '@aws-cdk/aws-ec2';
import { KubernetesVersion, EndpointAccess, Cluster, NodegroupAmiType, MachineImageType, ServiceAccount, HelmChart } from '@aws-cdk/aws-eks';
import { Role, AccountRootPrincipal, PolicyDocument, PolicyStatement, Effect, ServicePrincipal } from '@aws-cdk/aws-iam';
import { WordpressStackProps } from './wordpress-stack-props';


/**
 * Custom Properties for the EKS Stack
 */
 export interface EFSStackProps extends WordpressStackProps {
   efsId: string;
   efsSecurityGroup: SecurityGroup
}

export class EksStack extends Stack {

    private props: EFSStackProps;
    private cluster: Cluster;
    private clusterName: string;
    private serviceAccountName: string;
    private describeEfsPolicyStatement: PolicyStatement;
    private createEfsPolicyStatement: PolicyStatement;
    private deleteEfsPolicyStatement: PolicyStatement;
    constructor(scope: App, id: string, props: EFSStackProps) {
        super(scope, id, props);
        this.props = props;
        this.clusterName = 'k8s-' + this.props.environmentName;
        this.createPolicyStatements();
        this.createCluster()
        this.createAutoScalingGroup()
        this.createEFSServiceAccount()
        this.addCSIDriver()
        this.createStorageClass()
    }


    /**
     * Creates a EKS Cluster
     */
    private createCluster(): void {
        // cluster master role
        const masterRole = new Role(this, this.clusterName + '-master-role', {
            assumedBy: new AccountRootPrincipal(),
        });
        const vpc = Vpc.fromLookup(this,'vpc',{vpcName: 'vpc-' + this.props.environmentName})
        // Create a EKS cluster with Fargate profile.
        this.cluster = new Cluster(this, `eks-${this.clusterName}`, {
            version: KubernetesVersion.V1_21,
            mastersRole: masterRole,
            clusterName: this.clusterName,
            outputClusterName: true,
            defaultCapacity: 0,
            outputMastersRoleArn: true,
            endpointAccess: EndpointAccess.PUBLIC, // In Enterprise context, you may want to set it to PRIVATE.
            vpc,
            vpcSubnets: [{ subnetType: SubnetType.PRIVATE_WITH_NAT }]
        });


          
    }

    private createPolicyStatements() {
      this.describeEfsPolicyStatement = new PolicyStatement({
        resources: ['*'],
        actions: [
          'elasticfilesystem:DescribeAccessPoints',
          'elasticfilesystem:DescribeFileSystems'
        ],
        effect: Effect.ALLOW,
      })
      this.createEfsPolicyStatement = new PolicyStatement({
        resources: ['*'],
        actions: [
          'elasticfilesystem:CreateAccessPoint'
        ],
        effect: Effect.ALLOW,
      })
      this.deleteEfsPolicyStatement = new PolicyStatement({
        resources: ['*'],
        actions: [
          'elasticfilesystem:DeleteAccessPoint'
        ],          
        effect: Effect.ALLOW,
      });
      //const efsPolicyDocument = new PolicyDocument({ statements: [ this.describeEfsPolicyStatement, this.createEfsPolicyStatement, this.deleteEfsPolicyStatement ] });
    }

    private createAutoScalingGroup() {
      const autoScalingGroupName = `asg-k8s-${this.props.environmentName}`          
      const autoscalingGroup = this.cluster.addAutoScalingGroupCapacity(autoScalingGroupName,{
        instanceType: new InstanceType(this.node.tryGetContext('clusterInstanceType') || 't3.small'),
        autoScalingGroupName, 
        minCapacity: this.node.tryGetContext('clusterMinCapacity') || 2,
        maxCapacity: this.node.tryGetContext('clusterMaxCapacity') || 4,
        desiredCapacity: this.node.tryGetContext('clusterDesiredCapacity') || 2,
        machineImageType: MachineImageType.AMAZON_LINUX_2           
      })
      autoscalingGroup.scaleOnCpuUtilization('cpuUtilization', {
        targetUtilizationPercent: this.node.tryGetContext('clusterCpuUtilization') || 30,
        cooldown: Duration.minutes(this.node.tryGetContext('clusterCoolDownMinutes') || 2)
      })
      autoscalingGroup.addToRolePolicy
      autoscalingGroup.addSecurityGroup(this.props.efsSecurityGroup)

    }

    private createEFSServiceAccount(){

      // ðŸ‘‡ Create a Policy Document (Collection of Policy Statements)
      //
      this.serviceAccountName= `efs-sa`
      const serviceAccount = new ServiceAccount(this, this.serviceAccountName, {
        cluster: this.cluster,
        name: this.serviceAccountName,
        namespace: 'kube-system'
      })
      serviceAccount.addToPrincipalPolicy(this.describeEfsPolicyStatement)
      serviceAccount.addToPrincipalPolicy(this.createEfsPolicyStatement)
      serviceAccount.addToPrincipalPolicy(this.deleteEfsPolicyStatement)
    }


    private addCSIDriver() {

      new HelmChart(this, 'aws-efs-csi-driver', {
        cluster: this.cluster,
        chart: 'aws-efs-csi-driver',
        repository: 'https://kubernetes-sigs.github.io/aws-efs-csi-driver/',
        namespace: 'kube-system',
        release: 'v1.3.6',
        values: {
          'image.repository': `${this.props.env?.account}.dkr.ecr.${this.props.env?.region}.amazonaws.com/eks/aws-efs-csi-driver`,
          'controller.serviceAccount.create': false,
          'controller.serviceAccount.name': this.serviceAccountName
        }
      });

    }

    private createStorageClass() {

      const namespace = this.cluster.addManifest('storage-class', {
        apiVersion: 'storage.k8s.io/v1',
        kind: 'StorageClass',
        metadata: { 
          name: 'efs-sc',
          namespace: 'kube-system'
        },
        provisioner: 'efs.csi.aws.com',
        reclaimPolicy: 'Retain',
        volumeBindingMode: 'WaitForFirstConsumer',
        parameters: {
          'provisioningMode': 'efs-ap',
          'fileSystemId': this.props.efsId,
          'directoryPerms': '700',
          'gidRangeStart': '1000',
          'gidRangeEnd': '2000',
          'basePath': '/data'
        }
      });
    }

}
