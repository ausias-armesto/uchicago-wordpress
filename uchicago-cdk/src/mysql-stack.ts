import { Stack, Construct, Duration, Tags, RemovalPolicy } from '@aws-cdk/core'
import {
  Instance, InstanceType, MachineImage, CloudFormationInit, InitFile,
   IVpc, Vpc, SecurityGroup, InitCommand, Peer, Port, SubnetType
} from '@aws-cdk/aws-ec2'
import { KeyPair } from 'cdk-ec2-key-pair'
import { LogGroup, RetentionDays } from '@aws-cdk/aws-logs'
import { ARecord, HostedZone, RecordTarget } from '@aws-cdk/aws-route53';
import { WordpressStackProps } from './wordpress-stack-props';

/**
 * Custom Properties for the Mysql Stack
 */
 export interface MysqlStackProps extends WordpressStackProps {
  rootPassword: string;
  wordpressPassword: string;
}

/**
 * Construct all Mysql resources on AWS
 */
export class MysqlStack extends Stack {

  private props: MysqlStackProps;
  private keyPair: KeyPair
  private mysqlInstance: Instance;
  private vpc: IVpc;

  constructor(scope: Construct, id: string, props?: MysqlStackProps) {
    super(scope, id, props)
    this.props = props!;

    this.vpc = Vpc.fromLookup(this, 'vpc', { vpcName: 'vpc-' + this.props.environmentName })

    //this.createLogGroup()
    this.keyPair = this.createKeyPair()
    this.mysqlInstance = this.createEc2Instance()
    this.addDNS()

  }

  /**
   * Creates a LogGroup to send SSM Logs
   * @returns 
   */
  private createLogGroup(): LogGroup {
    return new LogGroup(this, 'log-group-ec2', {
      logGroupName: '/aws/ssm/managed-instances',
      retention: RetentionDays.ONE_MONTH,
      removalPolicy: RemovalPolicy.DESTROY
    })
  }

  /**
   * Creates a EC2 KeyPair to access the EC2 Instances
   * @returns 
   */
  private createKeyPair(): KeyPair {
    const name = `mysql-key-pair-${this.props.environmentName}`
    const key = new KeyPair(this, name, {
      name,
      description: `Key pair to access Mysql instance on ${this.props.environmentName} environment`,
      storePublicKey: true, // by default the public key will not be stored in Secrets Manager
    });
    return key
  }

  /**
   * Creates the conent of the initial SQL commands to execute the first time on the server
   */
   private createSqlContent() {
    return `
      CREATE USER 'wordpress'@'localhost' IDENTIFIED BY '${this.props.wordpressPassword}';
      GRANT ALL PRIVILEGES ON *.* TO 'wordpress'@'localhost' WITH GRANT OPTION;
      CREATE USER 'wordpress'@'10.0.%' IDENTIFIED BY '${this.props.wordpressPassword}';
      GRANT ALL PRIVILEGES ON *.* TO 'wordpress'@'10.0.%' WITH GRANT OPTION;
      CREATE DATABASE wordpress;
      FLUSH PRIVILEGES;
    `
  }

  /**
   * Creates the conent of the bash script
   */
   private createBashContent() {
    return `
    #!/bin/bash
    mysql --connect-expired-password -u root -p'${this.props.rootPassword}' < /tmp/mysql-init-config.sql
    `
  }  

  /**
   * Creates a EC2 instance with MySQL server
   * @returns 
   */
  private createEc2Instance() {
    const machineImage = MachineImage.lookup({
      name: 'uchicago-mysql-*',
      filters: { "tag:Name": ["wordpress-mysql-db"] }
    })
    const securityGroup = this.createSecurityGroup()
    const instanceType = new InstanceType(this.node.tryGetContext('mysqlInstanceType') || 't3.small' )
    const sqlFilepath = '/tmp/mysql-init-config.sql';
    const bashFilepath = '/tmp/mysql-init-config.sh';
    return new Instance(this, 'mysql-ec2-Instance', {
      vpc: this.vpc,
      // It should be in a PRIVATE_ISOLATED subnet, but Cloudformation needs the init scripts to communicate with Cloudformation to inform about the execution result status
      // and for doing that it needs internet connection
      vpcSubnets: this.vpc.selectSubnets({ subnetType: SubnetType.PRIVATE_WITH_NAT }),
      instanceType,
      machineImage,
      keyName: this.keyPair.keyPairName,
      securityGroup,
      init: CloudFormationInit.fromElements(
        InitFile.fromString(sqlFilepath, this.createSqlContent()),
        InitFile.fromString(bashFilepath, this.createBashContent()),
        InitCommand.argvCommand(['chmod', '+x', bashFilepath]),
        InitCommand.argvCommand(['bash', bashFilepath]),
        InitCommand.argvCommand( ['rm', '-rf', sqlFilepath, bashFilepath])
      ),
      initOptions: {
        // Optional, how long the installation is expected to take (5 minutes by default)
        timeout: Duration.minutes(30),
        // Optional, whether to include the --url argument when running cfn-init and cfn-signal commands (false by default)
        includeUrl: true,
        // Optional, whether to include the --role argument when running cfn-init and cfn-signal commands (false by default)
        includeRole: true,
      },
    });
  }

  /**
  * Creates a Security Group assigned to the Subnet created
  * @returns 
  */
  private createSecurityGroup(): SecurityGroup {
    const securityGroupName = 'secg-mysql-' + this.props.environmentName
    const sg = new SecurityGroup(this, securityGroupName, {
      securityGroupName: securityGroupName,
      vpc: this.vpc,
      description: 'Security Group for Mysql EC2 instance'
    })

    // Allow MySQL access
    sg.addIngressRule(Peer.ipv4('10.0.0.0/16'), Port.tcp(3306), 'Allow MySQL to VPC')
    sg.addIngressRule(Peer.ipv4('85.59.152.72/32'), Port.tcp(3306), 'Allow MySQL from home')

    // Access from home
    sg.addIngressRule(Peer.ipv4('85.59.152.72/32'), Port.tcp(22), 'Allow SSH access from home')

    Tags.of(sg).add('Name', securityGroupName)
    return sg
  }

  private addDNS() {
    const environmentHostedZone = HostedZone.fromLookup(this,'HostedZone',{
      domainName: `${this.props.environmentName}.wordpress.uchicago.local`,
      privateZone: true,
      vpcId: this.vpc.vpcId
    });

    const mysqlDnsEntry = new ARecord(this,'mysql-record-dns', {
      target: RecordTarget.fromIpAddresses(this.mysqlInstance.instancePrivateIp),
      recordName: 'mysql',
      zone: environmentHostedZone
    })
    
  }

}
