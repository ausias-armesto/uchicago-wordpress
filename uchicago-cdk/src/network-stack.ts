import { Stack, Construct, Tags } from '@aws-cdk/core'
import { Vpc, SubnetType } from '@aws-cdk/aws-ec2'
import { HostedZone } from '@aws-cdk/aws-route53';
import { WordpressStackProps } from './wordpress-stack-props';


/**
 * Construct all networking resources on AWS
 */
export class NetworkStack extends Stack {

  private props: WordpressStackProps;
  private vpc: Vpc

  constructor(scope: Construct, id: string, props?: WordpressStackProps) {
    super(scope, id, props)
    this.props = props!;
    this.vpc  = this.createVpc()
    this.createHostedZone()

  }

  /**
   * Creates the VPC, Subnets, routing tables, Internet Gateway
   * @returns 
   */
  private createVpc (): Vpc {
    const vpc = new Vpc(this, 'VPC', {
      cidr: '10.0.0.0/16',
      enableDnsHostnames: true,
      enableDnsSupport: true,
      maxAzs: 3,      
      subnetConfiguration: [
        {
          cidrMask: 24,
          name: 'Private-',
          subnetType: SubnetType.PRIVATE_WITH_NAT
        },
        {
          cidrMask: 24,
          name: 'Public-',
          subnetType: SubnetType.PUBLIC
        }                       
      ]
   });
   Tags.of(vpc).add('Name','vpc-' + this.props.environmentName)
   return vpc
  }

  private createHostedZone() {
    const environmentHostedZone = new HostedZone(this, 'HostedZone', {
      zoneName: `${this.props.environmentName}.wordpress.uchicago.local`,
      comment: `Hosted zone for Wordpress ${this.props.environmentName} environment`
    });
    environmentHostedZone.addVpc(this.vpc)
  }

}
