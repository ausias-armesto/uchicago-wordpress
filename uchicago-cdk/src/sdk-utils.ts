var AWS = require("aws-sdk");

export class SdkUtils {

  private secretsmanager: any;

  constructor() {
    AWS.config.getCredentials(function (err: any) {
      if (err) console.log(err.stack);
    });
    this.secretsmanager = new AWS.SecretsManager();
  }


  /**
   * Get a secret created on AWS
   * @param secretName The name of the secret
   * @returns The given password of the secret
   */
  public async getSecret(secretName: string): Promise<string> {
    var getSecretParams = {
      SecretId: secretName
    };    
    return new Promise((resolve, reject) => {
      this.secretsmanager.getSecretValue(getSecretParams, function (err: any, data: any) {
        if (err) {
          reject(err)
        } else { // Retrieving an existing secret          
          resolve(JSON.parse(data.SecretString).password)
        }
      });
    })
  }


}
