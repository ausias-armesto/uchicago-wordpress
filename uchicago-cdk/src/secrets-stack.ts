import { Stack, Construct, RemovalPolicy } from '@aws-cdk/core'
import { Secret } from '@aws-cdk/aws-secretsmanager';
import { WordpressStackProps } from './wordpress-stack-props';


/**
 * Construct all Secrets resources on AWS
 */
export class SecretsStack extends Stack {

  private props: WordpressStackProps;

  constructor(scope: Construct, id: string, props?: WordpressStackProps) {
    super(scope, id, props)
    this.props = props!;
    this.createSecret(`root-mysql-uchicago-${this.props.environmentName}`, 'root', `Root user for Mysql database on ${this.props.environmentName}`)
    this.createSecret(`wordpress-mysql-uchicago-${this.props.environmentName}`, 'root', `Wordpress user for Mysql database on ${this.props.environmentName}`)
    this.createSecret(`admin-wordpress-uchicago-${this.props.environmentName}`, 'admin', `Admin user for Wordpress App on ${this.props.environmentName}`)

  }

  /**
   * Creates a Secret
   * @returns 
   */
  private createSecret(secretName: string, user: string, description: string): Secret {
    return new Secret(this, secretName, {
      secretName: secretName,
      description: description,
      generateSecretString: {
        secretStringTemplate: JSON.stringify({ username: user }),
        generateStringKey: 'password',
        requireEachIncludedType: false,
        excludePunctuation: false,
        excludeCharacters: '"`\'\\'
      },
      removalPolicy: RemovalPolicy.DESTROY
    })
  }

}
