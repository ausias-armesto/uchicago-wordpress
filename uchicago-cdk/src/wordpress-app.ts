import { App } from '@aws-cdk/core'
import { NetworkStack } from './network-stack'
import { MysqlStack } from './mysql-stack'
import { EksStack } from './eks-stack'
import { SecretsStack } from './secrets-stack'
import { WordpressStackProps } from './wordpress-stack-props'
import { SdkUtils } from './sdk-utils'
import { EFSStack } from './efs-stack'

export class WordpressApp {

  private sdkUtils: SdkUtils;
  private app: App;
  private props: WordpressStackProps


  constructor() {
    this.app = new App()
    this.sdkUtils = new SdkUtils()
    this.props = {
      env: {
        account: process.env.CDK_DEFAULT_ACCOUNT,
        region: this.app.node.tryGetContext('region') || process.env.CDK_DEFAULT_REGION
      },
      environmentName: this.app.node.tryGetContext('environmentName')
    } as WordpressStackProps
  }

  /**
   * Build the CDK Application
   */
  public async buildCdkApp() {
    const stackName = this.app.node.tryGetContext('stackName')
    switch (stackName) {
      case 'secrets':
        new SecretsStack(this.app, `wordpress-secrets-${this.props.environmentName}`, this.props)
        break;
      case 'network':
        new NetworkStack(this.app, 'wordpress-vpc-' + this.props.environmentName, this.props)
        break;
      case 'mysql':
        await this.createMySQLStack()
        break;
      case 'efs':
        new EFSStack(this.app, 'wordpress-efs-' + this.props.environmentName, this.props)
        break;        
      case 'kubernetes':
        await this.createEKSStack()
        break;
      case 'infrastructure':
        const mysqlStackPromise = this.createMySQLStack()
        const eksStackPromise = this.createEKSStack()
        await Promise.all([mysqlStackPromise, eksStackPromise])
        break;
      default:
        throw new Error(`Incorrect stack name ${stackName}`)
    }
  }

  /**
   * Creates the EKS Stack
   */
  private async createEKSStack() {
    const efsStack = new EFSStack(this.app, 'wordpress-efs-' + this.props.environmentName, this.props)
    new EksStack(this.app, 'wordpress-k8s-' + this.props.environmentName, {...this.props, efsSecurityGroup: efsStack.securityGroup, efsId: efsStack.fileSystem.fileSystemId})
  }

  /**
   * Create the MySQL Stack
   */
  private async createMySQLStack() {
    const rootPassword = await this.sdkUtils.getSecret(`root-mysql-uchicago-${this.props.environmentName}`)
    const wordpressPassword = await this.sdkUtils.getSecret(`wordpress-mysql-uchicago-${this.props.environmentName}`)
    const mysqlStackProps = {...this.props, rootPassword, wordpressPassword }
    new MysqlStack(this.app, 'wordpress-mysql-' + this.props.environmentName, mysqlStackProps)
  }

}
