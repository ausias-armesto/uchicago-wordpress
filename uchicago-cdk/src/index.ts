#!/usr/bin/env node
import 'source-map-support/register'
import { WordpressApp } from './wordpress-app'

const wordpressApp: WordpressApp = new WordpressApp()
wordpressApp.buildCdkApp()