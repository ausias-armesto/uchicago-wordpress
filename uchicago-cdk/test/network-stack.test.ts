import { SynthUtils } from '@aws-cdk/assert'
import { App } from '@aws-cdk/core'
import { NetworkStack } from '../src/network-stack'

test('NetworkStack - Check against snapshot', () => {
  const app = new App()
  // WHEN
  const stack: NetworkStack = new NetworkStack(app, 'MyTestStack', {environmentName: 'test'})
  // THEN
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
})
