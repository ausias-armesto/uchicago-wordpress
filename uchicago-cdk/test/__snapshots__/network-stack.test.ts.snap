// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkStack - Check against snapshot 1`] = `
Object {
  "Resources": Object {
    "HostedZoneDB99F866": Object {
      "Properties": Object {
        "HostedZoneConfig": Object {
          "Comment": "Hosted zone for Wordpress test environment",
        },
        "Name": "test.wordpress.uchicago.local.",
        "VPCs": Array [
          Object {
            "VPCId": Object {
              "Ref": "VPCB9E5F0B4",
            },
            "VPCRegion": Object {
              "Ref": "AWS::Region",
            },
          },
        ],
      },
      "Type": "AWS::Route53::HostedZone",
    },
    "VPCB9E5F0B4": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "vpc-test",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "VPCIGWB7E252D3": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "vpc-test",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "VPCPrivateSubnet1DefaultRouteA61E68C4": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VPCPublicSubnet1NATGatewayB4C4E7BE",
        },
        "RouteTableId": Object {
          "Ref": "VPCPrivateSubnet1RouteTableA4D05B11",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VPCPrivateSubnet1RouteTableA4D05B11": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Private-Subnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VPCPrivateSubnet1RouteTableAssociationF21E8F03": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VPCPrivateSubnet1RouteTableA4D05B11",
        },
        "SubnetId": Object {
          "Ref": "VPCPrivateSubnet1SubnetE506D08F",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VPCPrivateSubnet1SubnetE506D08F": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private-",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Private-Subnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VPCPrivateSubnet2DefaultRoute07980EC6": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "VPCPublicSubnet2NATGateway46565FAA",
        },
        "RouteTableId": Object {
          "Ref": "VPCPrivateSubnet2RouteTableDE9508E9",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VPCPrivateSubnet2RouteTableAssociation4B8D7DE0": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VPCPrivateSubnet2RouteTableDE9508E9",
        },
        "SubnetId": Object {
          "Ref": "VPCPrivateSubnet2Subnet876D15F0",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VPCPrivateSubnet2RouteTableDE9508E9": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Private-Subnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VPCPrivateSubnet2Subnet876D15F0": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.1.0/24",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private-",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Private-Subnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VPCPublicSubnet1DefaultRoute2D37E7F3": Object {
      "DependsOn": Array [
        "VPCVPCGW99B986DC",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VPCIGWB7E252D3",
        },
        "RouteTableId": Object {
          "Ref": "VPCPublicSubnet1RouteTable3110968E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VPCPublicSubnet1EIP47D0526D": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VPCPublicSubnet1NATGatewayB4C4E7BE": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "VPCPublicSubnet1EIP47D0526D",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "VPCPublicSubnet1SubnetFCDAB5AC",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VPCPublicSubnet1RouteTable3110968E": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VPCPublicSubnet1RouteTableAssociation3D9DD352": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VPCPublicSubnet1RouteTable3110968E",
        },
        "SubnetId": Object {
          "Ref": "VPCPublicSubnet1SubnetFCDAB5AC",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VPCPublicSubnet1SubnetFCDAB5AC": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.2.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public-",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VPCPublicSubnet2DefaultRoute99D502C6": Object {
      "DependsOn": Array [
        "VPCVPCGW99B986DC",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "VPCIGWB7E252D3",
        },
        "RouteTableId": Object {
          "Ref": "VPCPublicSubnet2RouteTableFB415FB4",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "VPCPublicSubnet2EIPD218F052": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "VPCPublicSubnet2NATGateway46565FAA": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "VPCPublicSubnet2EIPD218F052",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "VPCPublicSubnet2Subnet8A692BE6",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "VPCPublicSubnet2RouteTableAssociation6C2E4702": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "VPCPublicSubnet2RouteTableFB415FB4",
        },
        "SubnetId": Object {
          "Ref": "VPCPublicSubnet2Subnet8A692BE6",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "VPCPublicSubnet2RouteTableFB415FB4": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "VPCPublicSubnet2Subnet8A692BE6": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.3.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public-",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
          Object {
            "Key": "Name",
            "Value": "MyTestStack/VPC/Public-Subnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "VPCVPCGW99B986DC": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "VPCIGWB7E252D3",
        },
        "VpcId": Object {
          "Ref": "VPCB9E5F0B4",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
}
`;
